CalibTracker/SiStripCommon/interface/poly.h

```diff
@@ -111,8 +110,8 @@ template<class T> poly<T> operator+ (const char* lhs, const poly<T>& rhs ) { ret
 template <class charT, class traits, class T> 
 inline
 std::basic_ostream<charT,traits>& operator<<(std::basic_ostream<charT,traits>& strm, const poly<T>& f) { 
-  BOOST_FOREACH(std::set<T> column, std::make_pair(f.begin_columns(),f.end_columns())) 
-    { strm << "( "; BOOST_FOREACH(T entry, column) strm << entry << ", "; strm << " )" << std::endl; }
+  for(auto const& column : std::make_pair(f.begin_columns(),f.end_columns())) 
```

Comment:
IIUC, begin and end methods are still required (e.g., I can't take a simple std::vector v and do ```for(auto x: make_pair(v.begin(), v.end()) )```).

is this just never instantiated/compiled or am I missing something?

Reply:
You are right of course.

Actually it's not only that the function never gets specialized, but the whole file never gets included anywhere in CMSSW. Maybe it should just be deleted?

Reply:
another option is to add an accessor to columns and use it in this range loop

Reply:
Yes, that's what I did now, ok since you agree this is a good solution I will just push that after I managed to somehow test this class.

Reply:
Hi Slava, let me come back to the suggestion of deleting it. I was using the original class and it's implemented operators a bit, and I only got segfaults and undefined behaviour. Here my testing code:

```C++
#include "Poly.h"
#include <iostream>

int main()
{

    Poly<int> poly1;

    //poly1 += 1;
    //poly1 += 2;
    //poly1 += 3;

    poly1 *= 1;
    poly1 *= 2;
    poly1 *= 5;

    Poly<int> poly2;

    //poly2 += 1;
    //poly2 += 2;
    //poly2 += 3;

    poly2 *= 1;
    poly2 *= 2;
    poly2 *= 5;

    Poly<int> poly3 = poly1 + poly2;
    poly3 *= poly2;

    bool foo = poly2 < poly3;

    std::cout << foo << std::endl;
    std::cout << poly3 << std::endl;

    return 0;
}
```

This segfaults. If you remove the comments on the 6 lines I commented out, you get undefined behaviour (non deterministic result). I don't know if you want to have a unused, uncommented and buggy class in the release?

Reply:
CalibTracker/SiStripCommon is in alca category. I can not make an authoritative statement on removal.

Reply:
@lpernie, do you have an opinion on that?

Reply:
It seems to me that indeed this can be safely removed.

Reply:
Thank you, there it goes!